preCICE-Ubuntu-20.04/install-basics.sh

```diff
@@ -0,0 +1,20 @@
+#!/bin/sh
+set -ex
+
+# We (may) need the multiverse repository for the VBox Guest Additions
+sudo apt-add-repository multiverse
+apt-get update
+sudo apt-get upgrade -qy
+
+# Install the Xfce desktop environment and basic applications
+sudo apt-get install -y xubuntu-core^
+sudo apt-get install -y thunar xfce4-terminal terminator bash-completion tree evince firefox firefox-locale-en
```

Comment:
Why not simply install the [minimal version of ubuntu desktop](https://packages.ubuntu.com/focal/ubuntu-desktop-minimal)?
This may be more familiar to users.

```
sudo apt-get install ubuntu-desktop-minimal
```

Reply:
I will try, but I am afraid that it will have much higher memory requirements.

Reply:
This definitely looks more attractive, but I tried it and I have the following concerns:

* Memory usage: Xfce is around 300-400MB idle, Gnome is around 800-900MB idle (this is only a quick test I did).
* Gnome needs definitely more video memory (this may not be a real issue)
* I had very high latency in the GUI with a default `ubuntu-desktop-minimal`, I fixed them by changing the display controller to VMSVGA. I am not sure what the problem is exactly (I don't see it in other VMs), but it seems to me that this may cause some unexpected issues.
* With only `ubuntu-desktop-minimal` installed, the disk usage is 4.3GB. The current Xfce variant with preCICE, Python, OpenFOAM, ParaView, MPI, PETSc, ... is 10GB. Not directly comparable, but the GNOME image size may become a bit higher.

I would stick to Xfce for our first attempt. In any case, it is a fully functional desktop environment.

Reply:
> I had very high latency in the GUI

Disabling animations makes the desktop more snappy. I didn't observe noteworthy lag though.

> I would stick to Xfce for our first attempt.

Sure! Let's stick to the resource friendly DE.

Reply:
I tried LXDE and it is considerably more snappy in the VM. :+1:

Reply:
> Disabling animations makes the desktop more snappy. I didn't observe noteworthy lag though.

But this is again another tweak we don't need with xfce.

> I tried LXDE and it is considerably more snappy in the VM. 

LXDE is maybe the lightest, but also quite primitive (and meant as that). I find Xfce being on the sweet spot for the use case we have.